xtask/src/main.rs

```diff
@@ -0,0 +1,364 @@
+    )?;
+
+    // Setup Python environment
+    setup_python_environment(&connectedhomeip_dir)?;
+
+    // Build chip-tool if not cached or force rebuild
+    let chip_tool_path = connectedhomeip_dir.join("out/host/chip-tool");
+    if !chip_tool_path.exists() || force_rebuild {
+        build_chip_tool(&connectedhomeip_dir)?;
+    } else {
+        println!("Using existing chip-tool build");
+    }
+
+    println!("ConnectedHomeIP environment setup completed successfully!");
+    Ok(())
+}
+
+async fn build_rs_matter(profile: &str, additional_features: &[String]) -> Result<()> {
+    let project_root = get_project_root()?;
+    let examples_dir = project_root.join("examples");
+
+    println!("Building rs-matter examples...");
+
+    let mut features = DEFAULT_FEATURES.iter().map(|s| s.to_string()).collect::<Vec<_>>();
+    features.extend_from_slice(additional_features);
+    let features_str = features.join(",");
+
+    let mut cmd = Command::new("cargo");
+    cmd.args(&["build", "--bin", "onoff_light"])
+        .arg("--features").arg(&features_str)
+        .current_dir(&examples_dir);
+
+    if profile == "release" {
+        cmd.arg("--release");
+    }
+
+    run_command(&mut cmd)?;
+    println!("rs-matter examples built successfully!");
+    Ok(())
+}
+
+async fn run_integration_tests(
+    tests: &[String],
+    profile: &str,
+    timeout: u64,
+    skip_build: bool,
+) -> Result<()> {
+    let project_root = get_project_root()?;
+    let connectedhomeip_dir = project_root.join("connectedhomeip");
+
+    // Verify ConnectedHomeIP is set up
+    if !connectedhomeip_dir.exists() {
+        return Err(anyhow!(
+            "ConnectedHomeIP not found. Run 'cargo xtask itest-setup' first."
+        ));
+    }
+
+    let chip_tool_path = connectedhomeip_dir.join("out/host/chip-tool");
+    if !chip_tool_path.exists() {
+        return Err(anyhow!(
+            "chip-tool not found. Run 'cargo xtask itest-setup' first."
+        ));
+    }
+
+    // Build rs-matter if requested
+    if !skip_build {
+        build_rs_matter(profile, &[]).await?;
+    }
+
+    // Determine which tests to run
+    let tests_to_run = if tests.is_empty() {
+        get_enabled_tests()
+    } else {
+        tests.to_vec()
+    };
+
+    if tests_to_run.is_empty() {
+        println!("No tests specified and no default tests enabled.");
+        return Ok(());
+    }
+
+    println!("Running integration tests: {:?}", tests_to_run);
+
+    // Setup environment variables
+    let rs_matter_data = tempfile::tempdir()
+        .context("Failed to create temporary directory for rs-matter data")?;
+    
+    // Clean up any existing data
+    if rs_matter_data.path().exists() {
+        std::fs::remove_dir_all(rs_matter_data.path())
+            .context("Failed to clean rs-matter data directory")?;
+    }
+
+    let profile_dir = if profile == "release" { "release" } else { "debug" };
+    let onoff_light_path = project_root
```

Comment:
I think this should rather be:
```rust
    let onoff_light_path = project_root
        .join("target")
        .join(profile_dir)
        .join("onoff_light");
```

Since the `examples` project is part of a workspace, its build artefacts do end up in the workspace-wide "target/debug|release" directory, isn't it?